src/Compilers/Core/Portable/Operations/ICollectionElementInitializerOperation.cs

```diff
@@ -28,11 +27,16 @@ public interface ICollectionElementInitializerOperation : IOperation
         /// <summary>
         /// Arguments passed to add method invocation.
         /// </summary>
-        ImmutableArray<IOperation> Arguments { get; }
+        ImmutableArray<IArgumentOperation> Arguments { get; }
 
         /// <summary>
-        /// Flag indicating if this is a dynamic invocation.
+        /// Flag indicating if this is a dynamic invocation. Will always be false.
         /// </summary>
+        /// <remarks>
+        /// This flag is obsolete, and dynamic collection initializers have been broken out into a separate interface,
+        /// <see cref="IDynamicCollectionElementInitializerOperation"/>.
+        /// </remarks>
+        [Obsolete]
```

Comment:
>[Obsolete] [](start = 8, length = 10)

Alternatively we could keep this node as is. It feels like splitting it is a major API change. We we unable to represent some dynamic cases before?

Reply:
No, we were not missing any dynamic cases before that we're getting with the new API

---
In reply to: [187732590](https://github.com/dotnet/roslyn/pull/26781#discussion_r187732590) [](ancestors = 187732590)